instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/handlers/action_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActionPack
+      module Handlers
+        # Action controller handler to handle the notification from ActiveSupport
+        class ActionController
+          # @param config [Hash] of instrumentation options
+          def initialize(config)
+            @config = config
+          end
+
+          # Invoked by ActiveSupport::Notifications at the start of the instrumentation block
+          #
+          # @param _name [String] of the Event (unused)
+          # @param _id [String] of the event (unused)
+          # @param payload [Hash] containing job run information
+          # @return [Hash] the payload passed as a method argument
+          def start(_name, _id, payload)
+            rack_span = OpenTelemetry::Instrumentation::Rack.current_span
+
+            # from rails 6.1, the request will be added to payload
+            request = payload[:request]
+            request = payload[:headers].instance_variable_get(:@req) if ::ActionPack.version < Gem::Version.new('6.1.0')
+
+            rack_span.name = "#{payload[:controller]}##{payload[:action]}" unless request.env['action_dispatch.exception']
```

Comment:
I would like for this to look like a templatized route if possible including the HTTP Method.  

Something like this:

```ruby
rake_span.name = "#{payload[:method]}.underscore} #{payload[:controller].underscore/#{payload[:action].underscore} "
```

Do you know if the rails route is included in the payload?

Also, are the controller and action missing in the event of an exception? I think I would like it to rename the span if they are present.

What happens in the event of a 404 route not found?

Reply:
> Do you know if the rails route is included in the payload?

Yes, but only in payload from finish (i.e. finish of the subscribe). In that case, something like
```ruby
def finish(_name, _id, payload)
  rack_span = OpenTelemetry::Instrumentation::Rack.current_span
  rack_span.name = "#{payload[:method]} #{payload[:controller]}/#{payload[:action]}" unless request.env['action_dispatch.exception']
end
```


> Also, are the controller and action missing in the event of an exception? I think I would like it to rename the span if they are present.

Yes, they are present; I can add them for sure. I am just uncertain about the initial intention behind `rack_span.name = "#{self.class.name}##{name}" unless request.env['action_dispatch.exception']`. I guess in original monkey patch require no exception so the `#{self.class.name}##{name}` will be valid string (e.g. not empty string).

> What happens in the event of a 404 route not found?

For 404, (e.g. curl 0.0.0.0:80/url_that_do_not_exist) there will be no modification on rack_span with this implementation.

We only subscribe the `process_action.action_controller`, and from [debug_exceptions.rb](https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L31-L34), if the url is not found, the `Constants::X_CASCADE` will be `pass`, which will directly raise `ActionController::RoutingError` without going to process_action (I think that's because the [show_exceptions.rb](https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/middleware/show_exceptions.rb#L31) treat it as non exception?).

Reply:
After digging more on actionpack, I think `process_action.action_controller` won't be invoke if it's 404 route not found error (based on [actionable_exceptions.rb](https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/middleware/actionable_exceptions.rb#L16))